modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java

```diff
@@ -1363,8 +1541,7 @@ public <C extends Collection<?>> C readCollection(MessageCollectionItemType item
      * @return Whether array was fully written.
      */
     boolean writeArray(Object arr, long off, int len, int bytes) {
-        assert arr != null;
-        assert arr.getClass().isArray() && arr.getClass().getComponentType().isPrimitive();
+        assert arr == null || arr.getClass().isArray() && arr.getClass().getComponentType().isPrimitive();
```

Comment:
I don't think this change was necessary

Reply:
It was, I pass "null" here for direct buffers

Reply:
Ok, didn't notice that, thanks for the explanation